infer/lib/python/inferlib/jwlib.py

```diff
@@ -182,13 +184,28 @@ def run(self):
                            self.suppress_warnings_out)]
 
             with tempfile.NamedTemporaryFile(
-                    mode='w',
-                    suffix='.out',
-                    prefix='javac_',
-                    delete=False) as file_out:
+                mode='w',
+                prefix='javac_args_',
+                delete=False) as command_line:
+
+                args = filter(lambda x: not is_special_arg(x), javac_cmd)
+                command_line.write(" ".join(args))
```

Comment:
One of the tests fails here, because the encoding of the tempfile has a wrong default (and `temfile.NamedTemporaryFile()` doesn't accept an `encoding=` argument in Python 2, sigh...):

```sh
$ ./infer/tests/build_systems/build_integration_tests.py -- utf8_in_pwd
[...]
Running Gradle integration test
Traceback (most recent call last):
  File "/home/jul/infer/infer/bin/infer", line 223, in <module>
    main()
  File "/home/jul/infer/infer/bin/infer", line 175, in main
    capture_exitcode = imported_module.gen_instance(args, cmd).capture()
  File "/home/jul/infer/infer/lib/python/inferlib/capture/gradle.py", line 87, in capture
    return util.run_compilation_commands(cmds, clean_cmd)
  File "/home/jul/infer/infer/lib/python/inferlib/capture/util.py", line 40, in run_compilation_commands
    if cmd.start() != os.EX_OK:
  File "/home/jul/infer/infer/lib/python/inferlib/jwlib.py", line 252, in start
    self._compile()
  File "/home/jul/infer/infer/lib/python/inferlib/jwlib.py", line 327, in _compile
    return self.javac.run()
  File "/home/jul/infer/infer/lib/python/inferlib/jwlib.py", line 192, in run
    command_line.write(" ".join(args))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u03b9' in position 76: ordinal not in range(128)
[...]
```

Changing this line to `command_line.write(utils.encode(" ".join(args)))` fixes it. Can you please amend?

If you have Buck installed you should be able to run `make test` to make sure the tests pass (they do after I made this change, FYI, so all is good).